eks/mng/scale/scale.go

```diff
@@ -0,0 +1,239 @@
+}
+
+// New creates a new Scaler.
+func New(cfg Config) Scaler {
+	cfg.Logger.Info("creating tester", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	return &tester{cfg: cfg}
+}
+
+type tester struct {
+	cfg Config
+}
+
+var reqID = ""
+
+func (ts *tester) Scale(mngName string, update *eksconfig.MNGScaleUpdate) (err error) {
+	cur, ok := ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName]
+	if !ok {
+		ts.cfg.Logger.Warn("MNG not found; failing update", zap.String("mng-name", mngName))
+		return fmt.Errorf("MNGs[%q] not found; failed to update", mngName)
+	}
+	if len(cur.ScaleUpdates) == 0 {
+		ts.cfg.Logger.Info("MNG scaler is not enabled; skipping update", zap.String("mng-name", mngName))
+		return nil
+	}
+	for i, scaleUpdate := range cur.ScaleUpdates {
+		if scaleUpdate.Created {
+			if i == len(cur.ScaleUpdates) - 1 {
+				ts.cfg.Logger.Info("All updates have been applied for this MNG", zap.String("mng-name", mngName))
+				return nil
+			}
+		}
+	}
+
+	if ts.cfg.EKSConfig.LogColor {
+		colorstring.Printf("\n\n[yellow]*********************************[default]\n")
+		colorstring.Printf("[light_green]MNGs[%q].Update[default]\n", mngName)
+	} else {
+		fmt.Printf("\n\n*********************************\n")
+		fmt.Printf("MNGs[%q].Update\n", mngName)
+	}
+
+	ts.cfg.Logger.Info("starting tester.Update", zap.String("tester", reflect.TypeOf(tester{}).PkgPath()))
+	update.Created = true
+	ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName] = cur
+	cur, _ = ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName]
+	ts.cfg.EKSConfig.Sync()
+
+	createStart := time.Now()
+	defer func() {
+		createEnd := time.Now()
+		update.TimeFrameCreate = timeutil.NewTimeFrame(createStart, createEnd)
+		ts.cfg.EKSConfig.AddOnManagedNodeGroups.MNGs[mngName] = cur
+		ts.cfg.EKSConfig.Sync()
+	}()
+
+	ts.cfg.Logger.Info("waiting before starting MNG update",
+		zap.String("cluster-name", ts.cfg.EKSConfig.Name),
+		zap.String("mng-name", mngName),
+		zap.Duration("initial-wait", update.InitialWait),
+	)
+	select {
+	case <-time.After(update.InitialWait):
+		ts.cfg.Logger.Info("waited, starting MNG scaler",
+			zap.String("cluster-name", ts.cfg.EKSConfig.Name),
+			zap.String("mng-name", mngName),
+			zap.Int("asg-min-size", cur.ASGMinSize),
+			zap.Int("asg-max-size", cur.ASGMaxSize),
+			zap.Int("asg-desired-capacity", cur.ASGDesiredCapacity),
+			zap.Int64("target-min-size", update.MinSize),
+			zap.Int64("target-max-size", update.MaxSize),
+			zap.Int64("target-desired-size", update.DesiredSize),
+		)
+	case <-ts.cfg.Stopc:
+		ts.cfg.Logger.Warn("MNG scaler wait aborted; exiting", zap.String("mng-name", mngName))
+		return errors.New("MNG scaler wait aborted")
+	}
+
+	// ref. https://docs.aws.amazon.com/cli/latest/reference/eks/update-nodegroup-config.html
+	var scaleErr = ts.scaleMNG(mngName, update)
+	if scaleErr != nil {
+		return scaleErr
+	}
+
+	ts.cfg.Logger.Info("completed MNG scaler",
+		zap.String("from", ts.cfg.EKSConfig.Parameters.Version),
```

Comment:
We are not doing the version upgrades... Can we get previous scale config from `cur` for `"from"` field? And use the `update` for `"to"` field?